server/container/guice/cassandra-guice/src/test/java/org/apache/james/DockerCassandraRule.java

```diff
@@ -41,7 +41,7 @@ private static boolean isBindingToEveryThing(Binding binding) {
         return bindingIp == null || bindingIp.equals("0.0.0.0");
     }
 
-    private PropertiesConfiguration getCassandraConfigurationForDocker() {
+    public PropertiesConfiguration getCassandraConfigurationForDocker() {
```

Comment:
You should I guess have a look to module override...  
What you describe I believe can not be done in guice.